test/e2e/singlecluster/pod_test.go

```diff
@@ -108,29 +106,123 @@ var _ = ginkgo.Describe("Pod groups", func() {
 						}))
 					}
 				}).Should(gomega.Succeed())
-				// Verify that the Workload finishes.
-				gomega.Eventually(func(g gomega.Gomega) {
-					var wl kueue.Workload
-					g.Expect(k8sClient.Get(ctx, g1Key, &wl)).To(gomega.Succeed())
-					g.Expect(apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadFinished)).
-						To(gomega.BeTrueBecause("it's finished"))
-				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+				util.ExpectWorkloadToFinish(ctx, k8sClient, gKey)
 			})
 
 			ginkgo.By("Deleting finished Pods", func() {
-				for _, p := range g1 {
+				for _, p := range group {
 					gomega.Expect(k8sClient.Delete(ctx, p)).To(gomega.Succeed())
 				}
 				gomega.Eventually(func(g gomega.Gomega) {
-					for _, p := range g1 {
+					for _, p := range group {
 						var pCopy corev1.Pod
 						g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(p), &pCopy)).To(testing.BeNotFoundError())
 					}
 					var wl kueue.Workload
-					g.Expect(k8sClient.Get(ctx, g1Key, &wl)).Should(testing.BeNotFoundError())
+					g.Expect(k8sClient.Get(ctx, gKey, &wl)).Should(testing.BeNotFoundError())
 				}, util.Timeout, util.Interval)
 			})
 		})
+
+		ginkgo.It("Should only admit a complete group", func() {
+			group := podtesting.MakePod("group", ns.Name).
+				Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"1s"}).
```

Comment:
Can we actually use even less time than 1s? 1s is not bad, but still wasteful if could be replaced with 1ms. 

Also, 1s could flake in some situations when the testing code performs some checks before the pod completes (it is probably not the case here, but using 1ms should eliminate the risk).